app/script/components/mentionSuggestions.js

```diff
@@ -0,0 +1,117 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+window.z = window.z || {};
+window.z.components = z.components || {};
+
+z.components.MentionSuggestions = class MentionSuggestions {
+  constructor(params) {
+    this.isVisible = ko.observable(false);
+    this.onSelectionValidated = params.onSelectionValidated || (() => {});
+    this.suggestions = params.suggestions;
+    this.targetInputSelector = params.targetInput;
+    this.targetInput = undefined;
+
+    this.positions = ko.observable({});
+
+    this.selectedSuggestionIndex = ko.observable(0);
+    this.selectedSuggestion = ko.pureComputed(() => {
+      const index = this.selectedSuggestionIndex();
+      const suggestions = this.suggestions();
+      return suggestions[index];
+    });
+
+    this.suggestions.subscribe(suggestions => {
+      const shouldBeVisible = suggestions.length > 0;
+      if (shouldBeVisible && !this.isVisible()) {
+        this.initList();
+      } else if (!shouldBeVisible && this.isVisible()) {
+        this.teardownList();
+      }
+      this.isVisible(shouldBeVisible);
```

Comment:
`this.selectedSuggestionIndex` should be updated to stay in the bounds of the array length.